android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/register/dao/FamilyRegisterDao.kt

```diff
@@ -0,0 +1,243 @@
+          filterByResourceTypeId(Task.SUBJECT, ResourceType.Group, family.logicalId)
+        }
+    )
+  }
+
+  override suspend fun countRegisterData(appFeatureName: String?): Long {
+    // TODO fix this workaround for groups count
+    return fhirEngine
+      .search<Group> { getRegisterDataFilters(FAMILY.name).forEach { filterBy(it) } }
+      .filter { it.active && !it.name.isNullOrEmpty() }
+      .size
+      .toLong()
+  }
+
+  private suspend fun loadFamilyHead(family: Group) =
+    family.managingEntity?.let { reference ->
+      fhirEngine
+        .search<RelatedPerson> {
+          filterByResourceTypeId(
+            RelatedPerson.RES_ID,
+            ResourceType.RelatedPerson,
+            reference.extractId()
+          )
+        }
+        .firstOrNull()
+        ?.let { relatedPerson ->
+          fhirEngine
+            .search<Patient> {
+              filterByResourceTypeId(
+                Patient.RES_ID,
+                ResourceType.Patient,
+                relatedPerson.patient.extractId()
+              )
+            }
+            .firstOrNull()
+        }
+    }
+
+  private suspend fun loadFamilyMemberRegisterData(memberId: String) =
+    fhirEngine.load(Patient::class.java, memberId).let { patient ->
+      val conditions = loadMemberCondition(patient.logicalId)
+      val carePlans = loadMemberCarePlan(patient.logicalId)
+
+      RegisterData.FamilyMemberRegisterData(
+        id = patient.logicalId,
+        name = patient.extractName(),
+        age = (patient.birthDate.daysPassed() / DAYS_IN_YEAR).toString(),
+        birthdate = patient.birthDate,
+        gender = patient.gender.display.first().toString(),
+        pregnant = conditions.hasActivePregnancy(),
+        isHead = patient.isFamilyHead(),
+        deathDate = patient.extractDeathDate(),
+        servicesDue = carePlans.filter { it.due() }.flatMap { it.activity }.size,
+        servicesOverdue = carePlans.filter { it.overdue() }.flatMap { it.activity }.size
+      )
+    }
+
+  private suspend fun loadFamilyMemberProfileData(memberId: String) =
+    fhirEngine.load(Patient::class.java, memberId).let { patient ->
+      val conditions = loadMemberCondition(patient.logicalId)
+      val carePlans = loadMemberCarePlan(patient.logicalId)
+      val tasks = loadMemberTask(patient.logicalId)
+      val flags = loadMemberFlags(patient.logicalId)
+
+      FamilyMemberProfileData(
+        id = patient.logicalId,
+        name = patient.extractName(),
+        age = patient.birthDate.toAgeDisplay(),
+        birthdate = patient.birthDate,
+        gender = patient.gender,
+        pregnant = conditions.hasActivePregnancy(),
+        isHead = patient.isFamilyHead(),
+        deathDate = patient.extractDeathDate(),
+        conditions = conditions,
+        flags = flags,
+        services = carePlans,
+        tasks = tasks
+      )
+    }
+
+  private suspend fun loadMemberCondition(patientId: String) =
+    fhirEngine.search<Condition> {
```

Comment:
We can use searchResourceFor() which is the built-in implementation from DefaultRepository.

Same with other similar methods